io/aligned_buf_pool.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/cleanup/cleanup.h>
+
+#include <deque>
+#include <memory>
+#include <optional>
+
+#include "io/io.h"
+
+namespace io {
+
+// Buffer aligned to 4kb that is optionally backed by a registered buffer (from register_buffers)
+struct AlignedBuf {
+  AlignedBuf(MutableBytes buf, std::optional<unsigned> registered_idx = std::nullopt)
+      : buf(buf), registered_idx(registered_idx) {
+  }
+
+  const MutableBytes buf;                        // buf, nbytes
+  const std::optional<unsigned> registered_idx;  // optional buf_index for prep_read/write_fixed
+};
+
+// Hands out cunks of a large registered buffer, falls back to allocating non-registered
+// chunk if no free space is available.
+struct AlignedBufPool {
+  const static size_t kAlign = 4096;
+
+  // Request aligned buffer. Returns buffer and cleanup that recycles the buffer once dropped.
+  static auto Request(size_t length) {
```

Comment:
the length is always aligned and you always return a buffer of the specified length. 
if you follow the rest of my suggestions, you can just return `unique_ptr<void*, Deleter>` and I think it's gonna be a more strightforward interface.